test/reviewController.test.js

```diff
@@ -124,4 +124,95 @@ describe('ReviewController API Tests', () => {
       expect(response.body.allProductReviews).toEqual([]);
     }
   });
+  //create a review
+  it('should create a review successfully', async () => {
+    await Review.deleteMany({});
+    const userCredentials = {
+      email: 'Emily@google.com',
+      password: 'secret',
+    };
+
+    const signedCookie = await loginAndReturnCookie(userCredentials);
+
+    const album = await Album.findOne({ albumName: 'Unique Album' });
+
+    const newReviewData = {
+      rating: 4,
+      title: 'Loved this review',
+      comment: 'Recommend it"',
+      user: user._id,
+      album: album._id,
+    };
+    const response = await request(app)
+      .post(`/api/v1/reviews/album/${album._id}`)
+      .set('Cookie', signedCookie)
+      .send(newReviewData);
+
+    expect(response.status).toBe(StatusCodes.CREATED);
+    expect(response.body).toHaveProperty('review');
+    // If no album was found
+    if (!album) {
+      expect(response.status).toBe(StatusCodes.NOT_FOUND);
+    }
+    //create second review for the same album
+    const secondReviewData = {
+      rating: 4,
+      title: 'Loved this review',
+      comment: 'Recommend it"',
+      user: user._id,
+      album: album._id,
+    };
+    const res = await request(app)
+      .post(`/api/v1/reviews/album/${album._id}`)
+      .set('Cookie', signedCookie)
+      .send(secondReviewData);
+    // this user already submitted a review for this album
+    expect(res.status).toBe(StatusCodes.CONFLICT);
+  });
+
+  //  update a review
+  it('should update a review successfully', async () => {
+    const userCredentials = {
+      email: 'Emily@google.com',
+      password: 'secret',
+    };
+
+    const signedCookie = await loginAndReturnCookie(userCredentials);
+
+    const existingReview = await Review.findOne({});
+    // If no existingReview was found
+    if (!existingReview) {
+      expect(response.status).toBe(StatusCodes.NOT_FOUND);
+    }
+
+    const updatedReviewData = {
+      rating: 1,
+      title: 'Don not buy this album',
+      comment: 'Awful"',
+    };
+    if (!updatedReviewData) {
+      expect(response.status).toBe(StatusCodes.BAD_REQUEST);
+    }
+
+    // Check if the user is the author of the review
+    const authenticatedUserId = 'mockUserId'; // Replace with actual authenticated user's ID
+
+    if (existingReview.user.toString() !== authenticatedUserId) {
+      // User is not the author of the review, respond with unauthorized error
+      expect(StatusCodes.UNAUTHORIZED).toBe(StatusCodes.UNAUTHORIZED); // //AKOS: this one works but I feel it's wrong, couldn't write here 'respone', the terminal was complaining
+    }
```

Comment:
Separate test please.

To test if we get `UNAUTHORIZED` when a user tries to update a review that doesn't belong to them, we need to first create a second user that can legitimately log in and receive a signed cookie. Then user that second user's signed cookie to make a request to the endpoint, using the `existingReview._id` that belongs to `Emily`. That should correctly simulate a user trying to update a review that doesn't belong to them.

Reply:
@akosasante thank you, Akos, done!